src/gfxinit.cpp

```diff
@@ -215,9 +214,8 @@ static void LoadSpriteTables()
 	FillGRFDetails(extra, false, BASESET_DIR);
 	ClrBit(extra->flags, GCF_INIT_ONLY);
 
-	extra->next = top;
-	master->next = extra;
-	_grfconfig = master;
+	_grfconfig.insert(_grfconfig.begin(), *extra);
```

Comment:
What about something like `GRFConfig *master = _grfconfig.emplace(_grfconfig.begin(), master_filename);` around like 197?
Saves a copy here and a manual delete later on.
And if/when tackling `GRFConfig *extra`, the `.c_str()` isn't needed anymore since `GRFConfig` accepts `std::string`.